tangelo/linq/circuit.py

```diff
@@ -415,17 +443,80 @@ def remove_small_rotations(circuit, param_threshold=0.05):
     rot_gates = {"RX", "RY", "RZ", "CRX", "CRY", "CRZ"}
     gates = [g for g in circuit._gates if not (g.name in rot_gates and abs(g.parameter) % (2*np.pi) < param_threshold)]
 
-    return Circuit(gates)
+    return Circuit(gates) if remove_qubits else Circuit(gates, n_qubits=circuit.width)
```

Comment:
It is a matter of preference (and I also don't recall the original implementation), but I would keep this logic inside the `simply` method. This would remove the `remove_qubits` argument in the `remove_small_rotations` and `remove_redundant_gates` functions. This would ensure that those functions only do 1 thing, i.e. remove small rotations or redundant gates (and would leave the qubit indices as is, thus facilitating debugging if needed).

Reply:
What is the default behaviour you would expect from these "low level functions": remove or not remove qubits ? If a user was to use them directly (not through `simplify`), would you then expect them to "adjust" the number of qubits manually afterwards by deleting or manually ensuring they are all still there ?

The behaviour so far for "low level" functions was removing qubits, since we did not enforce `n_qubits` and let them change lists of gates and rebuild circuits without specifying anything. We had a test confirming that.

Reply:
Moving this forward: 
 
 The past behaviour was to trim necessary qubits, and we were not telling the user. This means that they could end up with shorter bit strings than expected without knowing what happened, or get error from simulators because they now need to trim their Hamiltonian.
 
 I'd rather make this explicit for now at high and low level (this is easy to revert). Moreover the output is then ready to work with the trimming functions implemented by Erika a while back. I think overall this is safer for users and more consistent with our workflow.